lib/formatters/default.js

```diff
@@ -41,6 +43,11 @@ class DefaultPrinter {
         this.delegates.push(new SarifPrinter(printOptions));
         break;
       }
+      case 'json': {
```

Comment:
We originally avoided adding json here because we didn't want `--formatter json` to mean the same thing as `--json`. Have we changed our mind about that?

Reply:
Well as per our comment above

```
    // TODO: add 'json' format to the list of those below. The plan is to
    // modify the 'json' formatter to write to a file, similar to the 'sarif' formatter.
```

I was doing that. Should we not now?

Reply:
I agree that this satisfies that comment now, but I think I also wanted to rethink the specific format. 

@scalvert - Let's decouple this for now (e.g. remove this case), and tackle it in a dedicate PR. Specifically, we should either a) we should introduce a `--formatter json` that uses a different format or b) deprecate the `--json` flag and suggest folks use `--formatter json`. I think "a" is fine, but I'd like that in a separate PR so it has a dedicated changelog entry (since it will certainly affect most editor integrations).

Reply:
Agree. Removing.

Reply:
Done